src/FileList/FileExtInfoLoader.cc

```diff
@@ -680,41 +688,76 @@ void FileExtInfoLoader::AddShapeEntries(CARTA::FileInfoExtended& extended_info,
     extended_info.set_depth(depth);
     extended_info.set_stokes(stokes);
 
+    auto* axes_numbers_info = extended_info.mutable_axes_numbers();
+    // Change to 1-based axis indices
+    axes_numbers_info->set_dir_x(direction_axes[0] + 1);
+    axes_numbers_info->set_dir_y(direction_axes[1] + 1);
+    axes_numbers_info->set_spectral(spectral_axis + 1);
+    axes_numbers_info->set_stokes(stokes_axis + 1);
+    axes_numbers_info->set_depth(depth_axis + 1);
+
+    // Set axis types with respect to axis numbers 0~3
+    std::vector<std::string> axis_types(4, "NA");
+    for (int i = 0; i < extended_info.header_entries_size(); ++i) {
+        auto header_entry = extended_info.header_entries(i);
+        auto entry_name = header_entry.name();
+        if (entry_name.find("CTYPE") == 0) {
+            auto entry_value = header_entry.value();
+            if (!entry_value.empty()) {
+                entry_value = entry_value.substr(0, entry_value.find("-", 0));
+                if (entry_name.back() == '1') {
+                    axis_types[0] = entry_value;
+                } else if (entry_name.back() == '2') {
+                    axis_types[1] = entry_value;
+                } else if (entry_name.back() == '3') {
+                    axis_types[2] = entry_value;
+                } else if (entry_name.back() == '4') {
+                    axis_types[3] = entry_value;
+                }
+            }
+        }
+    }
+
+    // In case if the stokes axis name is not available from the header info
+    if (stokes_axis > -1 && axis_types[stokes_axis] == "NA") {
+        axis_types[stokes_axis] = "STOKES";
+    }
+
     // shape computed_entry
     std::string shape_string;
     switch (num_dims) {
         case 2:
-            shape_string = fmt::format("[{}, {}]", shape(0), shape(1));
+            shape_string = fmt::format("[{}, {}] ({}, {})", shape(0), shape(1), axis_types[0], axis_types[1]);
             break;
         case 3:
-            shape_string = fmt::format("[{}, {}, {}]", shape(0), shape(1), shape(2));
+            shape_string =
+                fmt::format("[{}, {}, {}] ({}, {}, {})", shape(0), shape(1), shape(2), axis_types[0], axis_types[1], axis_types[2]);
             break;
         case 4:
-            shape_string = fmt::format("[{}, {}, {}, {}]", shape(0), shape(1), shape(2), shape(3));
+            shape_string = fmt::format("[{}, {}, {}, {}] ({}, {}, {}, {})", shape(0), shape(1), shape(2), shape(3), axis_types[0],
+                axis_types[1], axis_types[2], axis_types[3]);
             break;
     }
     auto shape_entry = extended_info.add_computed_entries();
     shape_entry->set_name("Shape");
     shape_entry->set_value(shape_string);
     shape_entry->set_entry_type(CARTA::EntryType::STRING);
 
-    if (chan_axis >= 0) {
+    if (depth_axis >= 0) {
```

Comment:
Shouldn't this be `spectral_axis` since it is "Number of channels"? If the image is RA-Channel-Dec (as in issue title) then depth is Dec axis.  To support images with axes in any order, the backend should distinguish between channels and depth (spectral axis vs. z axis), and so should the frontend.